knowledge-graph/src/main/scala/io/renku/knowledgegraph/projects/datasets/tags/TagsFinder.scala

```diff
@@ -59,15 +59,16 @@ private class TagsFinderImpl[F[_]: Async: NonEmptyParallel: Logger: SparqlQueryT
     Prefixes of (renku -> "renku", schema -> "schema"),
     s"""|SELECT DISTINCT ?name ?startDate ?maybeDesc ?dsIdentifier
         |WHERE {
-        |  ?projId a schema:Project;
-        |          renku:projectPath '${criteria.projectPath}';
-        |          renku:hasDataset ?dsId.
-        |  ?dsId renku:slug '${criteria.datasetName}';
-        |        schema:identifier ?dsIdentifier.
-        |  ?eventId schema:about/schema:url ?dsId;
-        |           schema:name ?name;
-        |           schema:startDate ?startDate.
-        |  OPTIONAL { ?eventId schema:description ?maybeDesc }
+        |  Graph ?g { ?projId renku:projectPath '${criteria.projectPath}' }
```

Comment:
Yes, things will work that way but I've noticed that reaching directly the proper graph might be a bit better. In order to do so, you can construct the Project ResourceId and use it to select the graph. An example might be this: https://github.com/SwissDataScienceCenter/renku-graph/blob/30bba7f33b692ffe41a96f153c56d58f705208e6/knowledge-graph/src/main/scala/io/renku/knowledgegraph/projects/details/KGProjectFinder.scala#L59

Reply:
Thank you! I didn't realize I can create a resource-id from a path!